### ISSUE TITLE
Motion to rename "WindowedApp" to "GraphicalApp"

### ISSUE THREAD

**Issue by KageKirin** [bkaradzic/GENie#418]
Hi,

related to the documentation PR #417, I'd like to motion to rename `WindowedApp` to `GraphicalApp`.
(I'm preparing a PR for this right now).

I'm open for comments and opinions about this.

PRO:
- `"win*"` configuration term to only match "windows" (or "win32", "winstore" etc), not "WindowedApp"
- using `"vs*"` for targeting "windows" seems a bit hazarduous, as technically VS can use other compilers more freely these days. I.e. technically, I _could_ want to create VS projects using another compiler (e.g. android-clang, or some console-specific one) (and changes for this could go mostly into a customized version of BX's `toolchain.lua`).

CON:
- unless allowing "WindowedApp" as exposed alias, this will be a breaking change. (Then again, nothing that a short `sed` call couldn't fix).

Regards.

**Comment by KageKirin** [comment]
Ok, I made #419 as a WIP PR.

**Comment by bkaradzic** [comment]
cc @rhoot for 2nd opinion...

**Comment by rhoot** [comment]
While I agree with the sentiment (`win*` vs `vs*`), this would be a huge breaking change. Enough so that I think it'd be a bad idea. The alias might work though.

**Comment by bkaradzic** [comment]
> using "vs*" for targeting "windows" seems a bit hazarduous, as technically VS can use other compilers more freely these days.

Btw, just to clarify this. Using "vs*" means in VS IDE, it doesn't imply compiler. In bx toolchain you can see that you can specify `--vs=<compiler>` which would explicitly set compiler, or in some cases it's possible to combine platforms together in the same solution that would use 2 different compilers depending on configuration choosen.

**Comment by bkaradzic** [comment]
Are you trying to have one application that could be compiled as ConsoleApp or WindowedApp by using some configuration option?

Not quite clear what you're trying to achieve. And I never seen need to do any configuration based on type of app that might change. Usually it's one or another, and then you just apply settings that are appropriate.

**Comment by KageKirin** [comment]
I had a few places where `configuration` was set to match 'WindowedApp' explicitely, and those were fine.
Main reason for actually doing this, is that I got repeatedly bitten by my own ~~forgetfulness~~ setting of `win*` for windows only settings, rather than using `vs*` (reason for this is that wanted to be able build for windows using gmake+clang as well).

**Comment by bkaradzic** [comment]
Ideally GENie would have something like: operating system / compiler / arch / cpu / crt / build system / IDE, and then you would be able to filter based on any of those.

**Comment by KageKirin** [comment]
Sounds like a feasible idea.
Ideally, each 'category' would be prefixed to make filtering easier.
e.g. `os-windows`, `os-linux`, `compiler-gcc`, `compiler-clang`, `compiler-msvc`, `arch-x86`, `arch-x64`, `arch-arm32`, `arch-arm64`, `ide-vs2019`, `ide-xcode10` etc.

Maybe for genie2? As these changes would be breaking backwards compatibility.

**Comment by bkaradzic** [comment]
Yeah, just slightly different: `arch-32`, `arch-64`, `cpu-arm`, `cpu-x86`
